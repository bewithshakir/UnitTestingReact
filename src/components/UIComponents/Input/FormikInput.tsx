import { useField } from 'formik';

import Input, { InputProps } from "./Input";

export default function FormikInput(props: InputProps) {
    const [, meta] = useField(props as any);
    return <Input {...props}
        id={props.name}
        error={Boolean(meta.error && meta.touched)}
        helperText={meta.touched && meta.error ? meta.error : undefined}
    />;
}
