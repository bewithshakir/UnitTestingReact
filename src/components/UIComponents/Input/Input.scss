.MuiInputBase-input.Mui-disabled {
    background: #F7F7F7 0% 0% no-repeat padding-box;
    opacity: 1;
}

.MuiFormLabel-root{
    color: #404040 !important;
}

.MuiFormLabel-root.Mui-focused {
    color: #404040!important;
}

.MuiInputBase-input:invalid {
    box-shadow: none;
    border: 2px solid #DD1D21;
}

.MuiFormHelperText-error{
    color:#DD1D21 !important;
}

.super{
    font-size: 1em;
}