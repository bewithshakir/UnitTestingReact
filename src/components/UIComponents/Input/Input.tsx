import { Fragment } from 'react';
import { InputLabel, InputBase, FormControl, FormHelperText } from '@material-ui/core';
import './Input.scss';

interface props {
  name?: string;
  id?: string;
  width?: number | string;
  label: string;
  placeholder?: string;
  description?: string;
  disabled?: boolean;
  required?: boolean;
  autoComplete?: string;
  classes?: any;
  multiline?: boolean;
  type?: string;
  error?: boolean;
  value?: string;
  helperText?: string
  onChange?: (...args: any[]) => void;
  onBlur?: (...args: any[]) => void;
  onFocus?:(...args: any[]) => void;
  onClick?:(...args: any[]) => void;
  innerRef?:any;
  onKeyDown?:(...args: any[]) => void;
  inputProps?:any;
  endAdornment?:any;
  startAdornment?:any;
}

export default function Input (props: props) {

  return (
    <Fragment>
      <FormControl className='formInput'>
        <InputLabel shrink htmlFor={props.id} style={{ color: 'var(--Darkgray)' }} aria-labelledby={props.label} aria-required={props.required}>
          <b>{props.label.toUpperCase()}{props.required && props.label && (<span className='super'>*</span>)}</b>
        </InputLabel>
        <InputBase
          id={props.id}
          ref={props.innerRef} 
          style={{ width: props.width }}
          aria-describedby={props.description}
          name={props.name}
          value={props.value}
          disabled={props.disabled}
          placeholder={props.placeholder}
          aria-required={props.required}
          autoComplete={props.autoComplete}
          classes={props.classes}
          multiline={props.multiline}
          type={props.type}
          error={props.error}
          onChange={props.onChange}
          onBlur={props.onBlur}
          onClick={props.onClick}
          onKeyDown={props.onKeyDown}
          inputProps={props.inputProps}
          endAdornment={props.endAdornment}
          startAdornment={props.startAdornment}
        />
        {props.helperText && (
          <FormHelperText
            id={props.description}
            error={props.error}>
            {props.helperText}
          </FormHelperText>)}
      </FormControl>
    </Fragment>
  );
}

Input.defaultProps = {
  width: '100%',
  disabled: false,
  required: false,
  id: "input",
  autoComplete: "new-password",
  error: false,
};
