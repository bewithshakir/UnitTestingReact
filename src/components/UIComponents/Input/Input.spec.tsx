import { shallow } from 'enzyme';
import Input from './Input';


describe('renders the Input component', () => {
    const component = shallow(<Input label='Input' />);

    it('snapshot test', () => {
        expect(component).toMatchSnapshot();
    });
});

describe('renders the Input component disabled', () => {
    const component = shallow(<Input label='Input' disabled/>);

    it('snapshot test', () => {
        expect(component).toMatchSnapshot();
    });
});

describe('renders the Input component required', () => {
    const component = shallow(<Input label='Input' required />);

    it('snapshot test', () => {
        expect(component).toMatchSnapshot();
    });
});

describe('renders the Input component error helpertext', () => {
    const component = shallow(<Input label='Input' error helperText='Error' />);

    it('snapshot test', () => {
        expect(component).toMatchSnapshot();
    });
});

describe('renders the Input component  helpertext', () => {
    const component = shallow(<Input label='Input' helperText='Message' />);

    it('snapshot test', () => {
        expect(component).toMatchSnapshot();
    });
});