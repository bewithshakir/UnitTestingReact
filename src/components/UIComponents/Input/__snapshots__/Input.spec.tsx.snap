// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Input component disabled snapshot test for Input component disabled 1`] = `
<Fragment>
  <WithStyles(ForwardRef(FormControl))
    className="formInput"
  >
    <WithStyles(ForwardRef(InputLabel))
      aria-labelledby="Input"
      aria-required={false}
      htmlFor="input"
      shrink={true}
      style={
        Object {
          "color": "var(--Darkgray)",
        }
      }
    >
      <b>
        INPUT
      </b>
    </WithStyles(ForwardRef(InputLabel))>
    <WithStyles(ForwardRef(InputBase))
      aria-required={false}
      autoComplete="new-password"
      disabled={true}
      error={false}
      id="input"
      style={
        Object {
          "width": "100%",
        }
      }
    />
  </WithStyles(ForwardRef(FormControl))>
</Fragment>
`;

exports[`renders the Input component error and helpertext snapshot test for Input component with error and helpertext 1`] = `
<Fragment>
  <WithStyles(ForwardRef(FormControl))
    className="formInput"
  >
    <WithStyles(ForwardRef(InputLabel))
      aria-labelledby="Input"
      aria-required={false}
      htmlFor="input"
      shrink={true}
      style={
        Object {
          "color": "var(--Darkgray)",
        }
      }
    >
      <b>
        INPUT
      </b>
    </WithStyles(ForwardRef(InputLabel))>
    <WithStyles(ForwardRef(InputBase))
      aria-required={false}
      autoComplete="new-password"
      disabled={false}
      error={true}
      id="input"
      style={
        Object {
          "width": "100%",
        }
      }
    />
    <WithStyles(ForwardRef(FormHelperText))
      error={true}
    >
      Error
    </WithStyles(ForwardRef(FormHelperText))>
  </WithStyles(ForwardRef(FormControl))>
</Fragment>
`;

exports[`renders the Input component required snapshot test for Input component required 1`] = `
<Fragment>
  <WithStyles(ForwardRef(FormControl))
    className="formInput"
  >
    <WithStyles(ForwardRef(InputLabel))
      aria-labelledby="Input"
      aria-required={true}
      htmlFor="input"
      shrink={true}
      style={
        Object {
          "color": "var(--Darkgray)",
        }
      }
    >
      <b>
        INPUT
        <span
          className="super"
        >
          *
        </span>
      </b>
    </WithStyles(ForwardRef(InputLabel))>
    <WithStyles(ForwardRef(InputBase))
      aria-required={true}
      autoComplete="new-password"
      disabled={false}
      error={false}
      id="input"
      style={
        Object {
          "width": "100%",
        }
      }
    />
  </WithStyles(ForwardRef(FormControl))>
</Fragment>
`;

exports[`renders the Input component with helpertext snapshot test for Input component with helpertext 1`] = `
<Fragment>
  <WithStyles(ForwardRef(FormControl))
    className="formInput"
  >
    <WithStyles(ForwardRef(InputLabel))
      aria-labelledby="Input"
      aria-required={false}
      htmlFor="input"
      shrink={true}
      style={
        Object {
          "color": "var(--Darkgray)",
        }
      }
    >
      <b>
        INPUT
      </b>
    </WithStyles(ForwardRef(InputLabel))>
    <WithStyles(ForwardRef(InputBase))
      aria-required={false}
      autoComplete="new-password"
      disabled={false}
      error={false}
      id="input"
      style={
        Object {
          "width": "100%",
        }
      }
    />
    <WithStyles(ForwardRef(FormHelperText))
      error={false}
    >
      Message
    </WithStyles(ForwardRef(FormHelperText))>
  </WithStyles(ForwardRef(FormControl))>
</Fragment>
`;

exports[`renders the default Input component snapshot test for default Input 1`] = `
<Fragment>
  <WithStyles(ForwardRef(FormControl))
    className="formInput"
  >
    <WithStyles(ForwardRef(InputLabel))
      aria-labelledby="Input"
      aria-required={false}
      htmlFor="input"
      shrink={true}
      style={
        Object {
          "color": "var(--Darkgray)",
        }
      }
    >
      <b>
        INPUT
      </b>
    </WithStyles(ForwardRef(InputLabel))>
    <WithStyles(ForwardRef(InputBase))
      aria-required={false}
      autoComplete="new-password"
      disabled={false}
      error={false}
      id="input"
      style={
        Object {
          "width": "100%",
        }
      }
    />
  </WithStyles(ForwardRef(FormControl))>
</Fragment>
`;
